src/TKEBudgetTerms.jl

```diff
@@ -268,14 +292,22 @@ end
 Calculate the shear production rate in the `model`'s `z` direction, considering velocities
 `u`, `v`, `w` and background (or average) velocities `U`, `V` and `W`.
 """
-function ZShearProduction(model, u, v, w, U, V, W; location = (Center, Center, Center))
-    validate_location(location, "ZShearProduction")
-    return KernelFunctionOperation{Center, Center, Center}(shear_production_z_ccc, model.grid;
+function ZShearProductionRate(model, u, v, w, U, V, W; location = (Center, Center, Center))
+    validate_location(location, "ZShearProductionRate")
+    return KernelFunctionOperation{Center, Center, Center}(shear_production_rate_z_ccc, model.grid;
                                                            computed_dependencies=(u, v, w, U, V, W))
 end
 
-ZShearProduction(model; U=ZeroField(), V=ZeroField(), W=ZeroField(), kwargs...) =
-    ZShearProduction(model, model.velocities..., U, V, W; kwargs...)
+"""
+    $(SIGNATURES)
+
+Calculate the shear production rate in the `model`'s `z` direction. At least one of the mean 
+velocities `U`, `V` and `W` must be specified otherwise the output will be zero.
+"""
+function ZShearProductionRate(model; U=0, V=0, W=0, kwargs...)
```

Comment:
In principle it might be better to use `ZeroField` rather than `0` because can define `u - ZeroField() = u`. (However, I'm not sure that shortcut exists).

Reply:
I agree with you that `ZeroField()` is probably better. The reason I used `0` is that currently this isn't working:

```julia
julia> model.velocities.u - Oceananigans.Fields.ZeroField()
ERROR: type ZeroField has no field grid
Stacktrace:
 [1] getproperty(x::Oceananigans.Fields.ZeroField{Int64, 3}, f::Symbol)
   @ Base ./Base.jl:38
 [2] validate_grid(a::Field{Face, Center, Center, Nothing, RectilinearGrid{Float64, Periodic, Periodic, Bounded, Float64, Float64, Float64, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, CPU}, Tuple{Colon, Colon, Colon}, OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, Float64, FieldBoundaryConditions{BoundaryCondition{Oceananigans.BoundaryConditions.Periodic, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Periodic, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Periodic, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Periodic, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Flux, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Flux, Nothing}, BoundaryCondition{Oceananigans.BoundaryConditions.Flux, Nothing}}, Nothing, Oceananigans.Fields.FieldBoundaryBuffers{Nothing, Nothing, Nothing, Nothing}}, b::Oceananigans.Fields.ZeroField{Int64, 3})
```

But I think this should be implemented/fixed on the Oceananigans side, no?

Reply:
> In principle it might be better to use ZeroField rather than 0 because can define u - ZeroField() = u. (However, I'm not sure that shortcut exists).

@glwagner Do you have any reference that I can read about implementing such shortcut? I don't really know how to get started but I'd like to open a PR on Oceananigans to add that.